workflows4s-core/src/main/scala/workflows4s/wio/internal/EventEvaluator.scala

```diff
@@ -60,6 +60,17 @@ object EventEvaluator {
       doHandle(wio.eventHandler.map(_.asRight)).orElse(handleCheckpointBase(wio))
     }
 
+    override def onForEach[ElemId, InnerCtx <: WorkflowContext, ElemOut <: WCState[InnerCtx], InterimState <: WCState[Ctx]](
+        wio: WIO.ForEach[Ctx, In, Err, Out, ElemId, InnerCtx, ElemOut, InterimState],
+    ): Result = {
+      val state = wio.state(input)
+      val nexIndex: Int = GetIndexEvaluator.findMaxIndex(wio).map(_ + 1).getOrElse(index)
+      wio.eventEmbedding
+        .unconvertEvent(event)
+        .flatMap((elemId, convertedEvent) => new EventVisitor(state(elemId), convertedEvent, input, wio.initialElemState(), nexIndex).run.tupleLeft(elemId))
```

Comment:
Should we simplify `Embedded` as well? Instead of bidrectional support we jsut take initial and inner -> outer?